src/chunkserver/chunkserver-common/hdd_utils.cc

```diff
@@ -102,3 +107,328 @@ void hddRemoveChunkFromContainers(IChunk *chunk) {
 
 	gChunksMap.erase(chunkIter);
 }
+
+int chunkWriteCrc(IChunk *chunk) {
+	TRACETHIS();
+	assert(chunk);
+
+	chunk->owner()->setNeedRefresh(true);
+
+	uint8_t *crcData = gOpenChunks.getResource(chunk->metaFD()).crcData();
+
+	{
+		DiskWriteStatsUpdater updater(chunk->owner(), chunk->getCrcBlockSize());
+		ssize_t ret = chunk->owner()->writeCrc(chunk, crcData);
+
+		if (ret != static_cast<ssize_t>(chunk->getCrcBlockSize())) {
+			int errmem = errno;
+			safs_silent_errlog(LOG_WARNING,
+			                   "chunk_writecrc: file: %s - write error",
+			                   chunk->metaFilename().c_str());
+			errno = errmem;
+			updater.markWriteAsFailed();
+			return SAUNAFS_ERROR_IO;
+		}
+	}
+
+	HddStats::overheadWrite(chunk->getCrcBlockSize());
+	return SAUNAFS_STATUS_OK;
+}
+
+int hddIOEnd(IChunk *chunk) {
+	assert(chunk);
+	TRACETHIS1(c->chunkid);
+
+	if (chunk->wasChanged()) {
+		int status = chunkWriteCrc(chunk);
+		PRINTTHIS(status);
+
+		if (status != SAUNAFS_STATUS_OK) {
+			// FIXME(hazeman): We are probably leaking fd here.
```

Comment:
Its this old comment still applicable to the new implementation? Please review it and remove the comment if appropriate.

Reply:
Maybe is best to do it in other PR.